version: '3.8'

services:
  api:
    build: .
    image: ghcr.io/carlosanamurov/prueba_fino/fastapi-app:latest
    environment:
      - DB_HOST=mysql-container-1
      - DB_PORT=3306
      - DB_USER=root
      - DB_PASSWORD=${MYSQL_ROOT_PASSWORD}
      - DB_NAME=${MYSQL_DATABASE}
    depends_on:
      mysql-container-1:
        condition: service_healthy
    ports:
      - "8000:8000"
    networks:
      - backend_network

  mysql-container-1:
    image: mysql:8.0
    environment:
      MYSQL_ROOT_PASSWORD: ${MYSQL_ROOT_PASSWORD}
      MYSQL_DATABASE: ${MYSQL_DATABASE}
    volumes:
      - mysql_data:/var/lib/mysql
      - ./docker/mysql-config.cnf:/etc/mysql/conf.d/custom.cnf
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost"]
      interval: 5s
      timeout: 10s
      retries: 10
    networks:
      - backend_network

volumes:
  mysql_data:

networks:
  backend_network:
    driver: bridge